core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -1219,10 +1283,12 @@ public Frame decorrelateRel(LogicalJoin rel) {
       return null;
     }
 
-    final RelNode newJoin =
-        LogicalJoin.create(leftFrame.r, rightFrame.r,
-            decorrelateExpr(currentRel, map, cm, rel.getCondition()),
-            ImmutableSet.of(), rel.getJoinType());
+    final RelNode newJoin = relBuilder
+        .push(leftFrame.r)
+        .push(rightFrame.r)
+        .join(rel.getJoinType(), decorrelateExpr(currentRel, map, cm, rel.getCondition()),
+            ImmutableSet.<CorrelationId>of())
```

Comment:
Couldn't the type be inferred?

Reply:
Yes. I removed all <CorrelationId> from my change.